### PR TITLE
fix: Fix concatenation not allowing type number

### PR DESCRIPTION
**Pull request by joanamelo**
Currently this package does not allow numbers in parameters, if this is not a security/OS thing (similar packages allow it) I suggest we allow numbers too. Here is a working tested Pull Request.

### CODE REVIEW

File: src/index.js [hunk 1]
```diff
@@ -40,7 +40,7 @@ export default function urlJoin(...parts) {
 
     // Join parts
     const partsStr = parts
-    .filter((part) => typeof part === 'string')
```
Comment (satazor) [inline src/index.js:43]: Perhaps we should support more types. The filter here was just to remove `null/undefined` values. Having that said we can switch the filter to `part != null`.

Comment (satazor) [reply src/index.js:43]: The only downside is that `false` will be part of the URL if an argument is `false`, but I think thats ok.

**Review by satazor — CHANGES_REQUESTED** [review]

File: src/index.js (hunk 1 above)
Comment (joanamelo) [reply src/index.js:43]: If you pass an object or array the url will be invalid, right? I think it's best to keep it at string or number for the moment. What do you think?

Comment (satazor) [reply src/index.js:43]: Agree

**Review by satazor — APPROVED** [review]

**Pull request merged**